### PR TITLE
Configuration for running per-pull tests for kubernetes-sigs/cluster-api-provider-gcp

### PR DESCRIPTION
**Pull request by roberthbailey**
/assign @krousey @krzyzacy @BenTheElder

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    trigger: "/test( all| pull-cluster-api-provider-gcp-test)"
+    labels:
+      preset-service-account: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/cluster-api:v20180720-d52d72975
+        args:
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
```
Comment (krzyzacy) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: assume you want to execute something here?

Comment (roberthbailey) [reply config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: Yes, I was just thinking that....

Comment (roberthbailey) [reply config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: Ok, now actually executing something. Turns out that I needed the commands that had previously been hidden in `jobs/config.json`.

**Comment by krzyzacy** [comment]
/lgtm
/hold
`/hold cancel` whenever you want to deploy

**Comment by roberthbailey** [comment]
Let's do it now!

/hold cancel

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    context: pull-cluster-api-provider-gcp-build
+    rerun_command: "/test pull-cluster-api-provider-gcp-build"
+    trigger: "/test( all| pull-cluster-api-provider-gcp-build)"
+    labels:
+      preset-service-account: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/gcloud-in-go:v20171113-192bec25
+        args:	
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
```
Comment (krzyzacy) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: sigs.k8s.io, as this is not part of k8s.io... same as below

Comment (roberthbailey) [reply config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: Whoops. Sent you https://github.com/kubernetes/test-infra/pull/8762 to fix.

**Pull request merged**